### DISCUSSION TITLE
X Error of failed request:  BadLength (poly request too large or internal Xlib length error)

### DISCUSSION THREAD

**Discussion by antrrax** [israel-dryer/ttkbootstrap#147] [General]
I formatted the pc and installed linux mint 20.3 x64 Cinnamon X11 (Ubtuntu 20.4). and when trying to run an app with ttkbootstrap I got this error message:
```
X Error of failed request:  BadLength (poly request too large or internal Xlib length error)
   Major opcode of failed request:  139 (RENDER)
   Minor opcode of failed request:  20 (RenderAddGlyphs)
   Serial number of failed request:  792
   Current serial number in output stream:  801
```

To fix I had to install this package:
`sudo apt-get install fonts-symbola`

----

It would be interesting to put in the documentation that this font package is needed to use ttkbootstrap on ubuntu.

**Comment by israel-dryer** [comment]
I ran into this problem #8 early on and I tried to fix it, but ultimately decided to just remove glyphs that would cause these problems in built-in widget styles.

However, if you want to include emojis and other characters, it's best, as you said, to just install the supported font.

Tkinter uses the fonts installed on the machine, so there is really no way to package them with the application. That said... I did find this library the other day that I will have to look into.
https://github.com/TkinterEP/python-tkextrafont

As far as I know, ttkbootstrap doesn't _require_ symbola to run, but if you use certain glyphs in your application, it will crash tkinter. So, this is really dependent on the application that you build.

I ran the demo app on a very minimal repl.it linux distro with only the basic fonts available.
<img width="500" alt="image" src="https://user-images.githubusercontent.com/42584742/150577493-0befbd71-4c63-4479-8cfd-f1c6b83b1fad.png">

**Reply by israel-dryer** [reply]
As an additional note for future users, I've found that some "lite" versions of Linux are not packaged with fonts that include glyphs which include emojiis. So installing these system fonts are required in this case.
